
aruco_detector_node:
  ros__parameters:
    camera_frame: "Orca/camera_down_frame"

    # Flir camera
    subs:
      image_topic: "/cam_down/image_color"
      camera_info_topic: "/cam_down/camera_info"

    # Flir camera
    camera:  
      intrinsic: [1050.0, 1050.0, 960.0, 540.0] # fx, fy, cx, cy
      distortion: [-0.356890438215317, 0.168613819658546, 0.0, 0.0, -0.0441848241077346] # k1, k2, p1, p2, k3

    detect_board: true  # Set to true to detect aruco boards
    visualize: true  # Set to true to visualize the detected markers and board
    log_markers: false  # Set to true to log the detected markers

      # TAC ARUCO PARAMETERS
    aruco:
      marker_size: 0.150
      dictionary: "DICT_ARUCO_ORIGINAL"
    

    #TAC ARUCO BOARD PARAMETERS
    # board:
    #   xDist: 0.430
    #   yDist: 0.830
    #   ids: [28, 7, 96, 19]
    # marker_size: 0.150

    # Vortex docking plate
    board:
      xDist: 0.462
      yDist: 0.862
      ids: [28, 7, 96, 19]
    # aruco:
    #   marker_size: 0.167

    models:
      dynmod_stddev: 0.01
      sendmod_stddev: 0.01

